api/src/user/controllers/role.controller.ts

```diff
@@ -147,12 +153,26 @@ export class RoleController extends BaseController<
   @CsrfCheck(true)
   @Delete(':id')
   @HttpCode(204)
-  async deleteOne(@Param('id') id: string) {
-    const result = await this.roleService.deleteOne(id);
-    if (result.deletedCount === 0) {
-      this.logger.warn(`Unable to delete Role by id ${id}`);
-      throw new NotFoundException(`Role with ID ${id} not found`);
+  async deleteOne(@Param('id') id: string, @Req() req: Request) {
+    const userRoles = (req.user as User).roles;
+
+    const associatedUser = await this.userService.findOne({
+      roles: { $in: [id] },
+    });
+    if (userRoles.includes(id)) {
+      throw new ForbiddenException("Your account's role can't be deleted");
+    } else if (associatedUser) {
+      throw new ForbiddenException('Role is associated with other users');
+    } else {
+      try {
+        const result = await this.roleService.deleteOne(id);
+        if (result.deletedCount === 0) {
+          throw new NotFoundException(`Role with ID ${id} not found`);
+        }
+        return result;
+      } catch (error) {
+        throw new NotFoundException(`Role with ID ${id} not found`);
+      }
```

Comment:
You don’t need both checks. The second check already looks for all users, including the active one, so the first check is unnecessary. You can simplify by just using the second check to cover everything.

Reply:
Actually he doesn't need to know that he is the user using the role. The only information he needs is that some user is still using the role. 
This is still a suggestion for the behaviour, we have to check with @marrouchi  if this is the right think to do